test/fgpu/test_send.py

```diff
@@ -0,0 +1,256 @@
+################################################################################
+# Copyright (c) 2023, National Research Foundation (SARAO)
+#
+# Licensed under the BSD 3-Clause License (the "License"); you may not use
+# this file except in compliance with the License. You may obtain a copy
+# of the License at
+#
+#   https://opensource.org/licenses/BSD-3-Clause
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+"""Unit tests for :mod:`katgpucbf.fgpu.send`."""
+
+import asyncio
+from unittest import mock
+
+import aiokatcp
+import numpy as np
+import pytest
+import spead2.recv.asyncio
+import spead2.send.asyncio
+from katsdptelstate.endpoint import endpoint_list_parser
+
+from katgpucbf import N_POLS
+from katgpucbf.fgpu import METRIC_NAMESPACE
+from katgpucbf.fgpu.send import Chunk, make_descriptor_heap, make_streams
+from katgpucbf.utils import TimeConverter, gaussian_dtype
+
+from .. import PromDiff, unpack_complex
+
+pytest_mark = pytest.mark.parametrize("bits", [4, 8])
+N_SUBSTREAMS = 16
+N_CHUNKS = 5
+N_FRAMES = 7  # frames per chunk
+N_CHANNELS = 1024
+N_SPECTRA_PER_HEAP = 32  # Small to make the test fast
+SPECTRA_SAMPLES = 2 * N_CHANNELS
+FENG_ID = 3
+INTERFACES = ["10.0.0.1", "10.0.0.2"]
+NAME = "foo"
```

Comment:
Bikeshed:
Since we're writing in Python, this should really be something like "spam".

Reply:
There are already 22 other instances of "foo" in katgpucbf and 0 of "spam", so I'm sticking with "foo".

Reply:
Touche. I'll allow it this time. ;)